client/src/components/ShelfSelector.js

```diff
@@ -0,0 +1,279 @@
+
+  function handleBookShelfChange(event) {
+    const shelfIndex = shelves.findIndex(
+      (shelf) => shelf.id === event.target.value
+    );
+    shelfIndex >= 0 ? setShelfIndex(shelfIndex) : setShelfIndex(null);
+    setSelection({
+      ...selection,
+      bookshelf: event.target.value,
+      column: '',
+      compartment: '',
+    });
+    toast.dismiss('validationError');
+  }
+
+  function handleColumnChange(event) {
+    const columnIndex = shelves[shelfIndex].columns.findIndex(
+      (column) => column.id === event.target.value
+    );
+    columnIndex >= 0 ? setColumnIndex(columnIndex) : setColumnIndex(null);
+    setSelection({ ...selection, column: event.target.value, compartment: '' });
+    toast.dismiss('validationError');
+  }
+
+  function handleCompartmentChange(event) {
+    setSelection({ ...selection, compartment: event.target.value });
+    toast.dismiss('validationError');
+  }
+
+  function handleSelectionSave(event) {
+    event.preventDefault();
+    if (validateShelfSelection(selection)) {
+      onSelectShelf(selection, book);
+      onSetIsSelector(false);
+      toast.success('Book added to Shelf!', {
+        position: toast.POSITION.BOTTOM_CENTER,
+      });
+    } else {
+      toast.error('Please choose a shelf, column and compartment', {
+        position: toast.POSITION.BOTTOM_CENTER,
+        toastId: 'validationError',
+      });
+    }
+  }
+
+  return (
+    <ShelfSelectorCard isStatic={isStatic}>
+      <CloseButton
+        src={CloseIcon}
+        alt="Close Icon"
+        onClick={() => onSetIsSelector(false)}
+      />
+      <BookInformation>
+        <BookImageWrapper>
+          <img
+            src={
+              book.volumeInfo?.imageLinks?.thumbnail ||
+              book.volumeInfo?.imageLinks?.smallThumbnail
+            }
+            alt={book.volumeInfo?.title || 'Book Cover'}
+            width="128"
+            height="192"
+          />
+        </BookImageWrapper>
+
+        <BookSpecs>
+          <BookTitle>{book.volumeInfo?.title}</BookTitle>
+          {book.volumeInfo.subtitle && (
+            <BookSubTitle>{book.volumeInfo.subtitle}</BookSubTitle>
+          )}
+          <p>{book.volumeInfo?.authors?.[0]}</p>
+          <p>Released: {book.volumeInfo?.publishedDate?.substring(0, 4)}</p>
+          <p>ISBN: {book.volumeInfo?.industryIdentifiers[0]?.identifier}</p>
+        </BookSpecs>
+      </BookInformation>
+      <ShelfSelectorForm onSubmit={handleSelectionSave}>
+        <ShelfPicker>
+          <div>
+            <label htmlFor="bookshelf">Bookshelf</label>
+            <select
+              name="bookshelf"
+              id="bookshelf"
+              onChange={handleBookShelfChange}
+              value={selection.bookshelf}
+            >
+              <option value="">-Select-</option>
+              {shelves.length > 0 &&
+                shelves.map((shelf, index) => (
+                  <option key={index} value={shelf.id}>
```

Comment:
The shelf id would be better suitable for the key attribute. The same goes for the other select options.